crowbar_framework/app/models/service_object.rb

```diff
@@ -389,7 +389,7 @@ def select_nodes_for_role(all_nodes, role, preferred_intended_role = nil)
       valid_nodes = preferred_all_nodes unless preferred_all_nodes.empty?
     end
 
-    if role_constraints[role] && role_constraints[role].has_key?("count") && role_constraints[role]["count"] >= 0
+    if role_constraints[role] && role_constraints[role].key?("count") && role_constraints[role]["count"] >= 0
```

Comment:
Metrics/LineLength: Line is too long. [109/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)